foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/internal/helper/Helper.java

```diff
@@ -2074,29 +2075,24 @@ public static void systemBug(String description) {
             throw ConversionException.incorrectTimestampFormat(timestampString);
         }
 
-        try {
-            String nanoToken = timestampStringTokenizer.nextToken();
-            nanos = Integer.parseInt(nanoToken);
-            for (int times = 0; times < (9 - nanoToken.length()); times++) {
-                nanos = nanos * 10;
+        if (containsNanos) {
```

Comment:
Because of the added '+' with the tokenizer there may be a token representing the timezone after the nanos. If the nanos are missing the timezone may be interpreted as nanos. So the obsolete part probably is the `NoSuchElementException`. Which if deleted will leave the nanos uninitialized.